python/python/tests/test_transaction.py

```diff
@@ -1069,3 +1071,70 @@ async def test_lolwut_transaction(self, glide_client: GlideClusterClient):
         for element in results:
             assert isinstance(element, bytes)
             assert b"Redis ver. " in element
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_transaction_dump_restore(self, redis_client: TGlideClient):
+        cluster_mode = isinstance(redis_client, GlideClusterClient)
```

Comment:
You can use the parameterized cluster_mode arg instead of doing an isinstance check
```suggestion
    async def test_transaction_dump_restore(self, redis_client: TGlideClient, cluster_mode):
```